guides/micronaut-k8s/micronaut-k8s.adoc

```diff
@@ -0,0 +1,880 @@
+
+:exclude-for-build:gradle
+
+[source,bash]
+----
+./mvnw package -Dpackaging=docker
+----
+:exclude-for-build:
+
+:exclude-for-build:maven
+
+[source,bash]
+----
+./gradlew dockerBuild
+----
+:exclude-for-build:
+
+Edit `k8s.yml` inside `orders` service.
+
+[source, yaml]
+./orders/k8s.yml
+----
+include::{sourceDir}/micronaut-k8s/@sourceDir@/orders/k8s.yml[]
+----
+<1> Service name that we created in auth.yaml.
+<2> Name of the image that deployment use.
+<3> The imagePullPolicy is set to Never. We will always use local one that we built in previous step.
+<4> Name of a service, required for service discovery.
+<5> Micronaut default port on which application is running.
+
+Run next command to create described resources:
+
+[source,bash]
+----
+kubectl create -f orders/k8s.yml
+----
+
+[source,text]
+----
+deployment.apps/orders created
+service/orders created
+----
+
+=== API (Gateway) Microservice
+
+The next command will build docker image of `api` service with image name `api`.
+
+:exclude-for-build:gradle
+
+[source,bash]
+----
+./mvnw package -Dpackaging=docker
+----
+:exclude-for-build:
+
+:exclude-for-build:maven
+
+[source,bash]
+----
+./gradlew dockerBuild
+----
+:exclude-for-build:
+
+Edit `k8s.yml` inside `api` service.
+
+[source, yaml]
+./api/k8s.yml
+----
+include::{sourceDir}/micronaut-k8s/@sourceDir@/api/k8s.yml[]
+----
+<1> Service name that we created in auth.yaml.
+<2> Name of the image that deployment use.
+<3> The imagePullPolicy is set to Never. We will always use local one that we built in previous step.
+<4> Name of a service, required for service discovery.
+<5> Micronaut default port on which application is running.
+
+Run next command to create described resources:
+
+[source,bash]
+----
+kubectl create -f api/k8s.yml
+----
+
+[source,text]
+----
+deployment.apps/api created
+service/api created
+----
+
+=== Test integration between applications
+
+Run next command to check status of created pods and make sure that all have status "Running":
+
+[source,bash]
+----
+kubectl get pods
+----
+
+[source,text]
+----
+NAME                      READY   STATUS    RESTARTS   AGE
+api-774fd667b9-dmws4      1/1     Running   0          24s
+orders-74ff4fcbc4-dnfbw   1/1     Running   0          19s
+users-9f46dd7c6-vs8z7     1/1     Running   0          13s
+----
+
+Run next command to check status of created services:
+
+[source,bash]
+----
+kubectl get services
+----
+
+[source,text]
+----
+NAME         TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)          AGE
+api          NodePort    10.102.170.41    <none>        8080:32757/TCP   28m
+kubernetes   ClusterIP   10.96.0.1        <none>        443/TCP          11d
+orders       NodePort    10.107.116.38    <none>        8080:30534/TCP   28m
+users        NodePort    10.100.100.112   <none>        8080:31244/TCP   27m
+----
+
+Run next command to get url of api service:
+
+[source,bash]
+----
+export API_URL=$(minikube service api --url)
+----
```

Comment:
Not sure in Minikube, but in other distros, creating the service as `LoadBalancer` will expose the service in localhost. Please check if this is still needed.

Reply:
I added note how you can expose ip if someone wants, by default it will be in pending state.